### PR TITLE
CRA

### PR DESCRIPTION
**Pull request by celek**
New CRA tasks

### CODE REVIEW

**Comment by celek** [comment]
@padraic-edwards we are good to pull this one and test - I also pushed the bundles (tasks) to production - I will ping you to test

File: cra/sample/pr-pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+      description: the git repo
+    - name: branch
+      description: branch
+    - name: target-branch
+      description: target branch
+    - name: api-key
+      description: The IBM Cloud API key is used to access the IBM Cloud Kubernetes
+        Service API and interact with the cluster. You can obtain your API key with
+        'bx iam api-key-create' or via the console at https://cloud.ibm.com/iam#/apikeys
+        by clicking **Create API key** (Each API key only can be viewed once).
```
Comment (jauninb) [inline cra/sample/pr-pipeline.yaml:18]: where is api-key coming from ? I do not see it defined in the trigger-template

File: cra/sample/pr-pipeline.yaml [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+        - name: artifacts
+          workspace: artifacts
+        - name: secrets
+          workspace: artifacts
+      params:
+        - name: repository
+          value: $(params.repository)
+        - name: repository-integration
+          value: $(params.repository-integration)
+        - name: ibmcloud-api-key-secret-key
```
Comment (jauninb) [inline cra/sample/pr-pipeline.yaml:54]: this seems that api-key is the "key" that will/should contain the api key value in the CD Tekton Pipeline environment properties ?
Correct ?

File: cra/sample/pr-pipeline.yaml [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+        - name: repository
+          value: $(params.repository)
+        - name: revision
+          value: $(params.branch)
+        - name: commit-id
+          value: $(params.commit-id)
+        - name: commit-timestamp
+          value: $(params.commit-timestamp)
+        - name: ibmcloud-api-key
+          value: $(params.api-key)
```
Comment (jauninb) [inline cra/sample/pr-pipeline.yaml:97]: Hum... are we dealing here with the real value for api key to use ? not the indirection in the secret secure-properties by default ?

File: cra/sample/pr-pipeline.yaml [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+        - name: revision
+          value: $(params.branch)
+        - name: pr-url
+          value: $(params.pr-url)
+        - name: commit-id
+          value: $(params.commit-id)
+        - name: git-api-token-key
+          value: git-token
+        - name: ibmcloud-api-key
+          value: $(params.api-key)
```
Comment (jauninb) [inline cra/sample/pr-pipeline.yaml:147]: Hum... are we dealing here with the real value for api key to use ? not the indirection in the secret secure-properties by default ?

File: cra/sample/pr-pipeline.yaml [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+        - name: revision
+          value: $(params.branch)
+        - name: pr-url
+          value: $(params.pr-url)
+        - name: commit-id
+          value: $(params.commit-id)
+        - name: directory-name
+          value: $(tasks.code-fetch-code.results.directory-name)
+        - name: ibmcloud-api-key
+          value: $(params.api-key)
```
Comment (jauninb) [inline cra/sample/pr-pipeline.yaml:221]: Hum... are we dealing here with the real value for api key to use ? not the indirection in the secret secure-properties by default ?

File: cra/sample/pr-pipeline.yaml [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+        - name: pr-url
+          value: $(params.pr-url)
+        - name: commit-id
+          value: $(params.commit-id)
+        - name: target-branch
+          value: $(params.target-branch)
+        - name: target-commit-id
+          value: $(params.target-commit-id)      
+        - name: ibmcloud-api-key
+          value: $(params.api-key)
```
Comment (jauninb) [inline cra/sample/pr-pipeline.yaml:297]: Hum... are we dealing here with the real value for api key to use ? not the indirection in the secret secure-properties by default ?

File: cra/sample/pr-trigger.yaml [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+      type: kubernetes.io/dockerconfigjson
+    - apiVersion: v1
+      kind: ServiceAccount
+      imagePullSecrets:
+        - name: artifactory-pull-secret
+      metadata:
+        name: sa-pullsecret
+    - apiVersion: v1
+      kind: Secret
+      metadata:
```
Comment (jauninb) [inline cra/sample/pr-trigger.yaml:67]: is a specific/deciated secret required in the sample ?
why do not use the secure-properties secret that is created for us by CD Tekton Support - cf https://cloud.ibm.com/docs/ContinuousDelivery?topic=ContinuousDelivery-tekton_environment#tekton_envprop

**Review by jauninb — COMMENTED** [review]
It is not obvious to me how the api key value are passed along / managed in triggertemplate/pipeline and down to tasks
A sample README.md would help

File: cra/task-discovery.yaml [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+      description: git commit timestamp
+    - name: ibmcloud-api-key
+      description: the IBM Cloud API key
+      default: ""
+    - name: pipeline-debug
+      description: Pipeline debug mode
+      default: "1"
+    - name: directory-name
+      description: directory name where the repository is cloned
+    - name: pipeline-debug
```
Comment (padraic-edwards) [inline cra/task-discovery.yaml:25]: Duplicate pipeline-debug param.

Comment (celek) [reply cra/task-discovery.yaml:25]: fixed in next release

**Pull request closed without merging**